### PR TITLE
BF-196 disable checkout by sku

### PR DESCRIPTION
**Pull request by aleksa-andersinnovation**

### CODE REVIEW

File: Block/Checkout/Overview.php [hunk 1]
```diff
@@ -117,8 +118,9 @@ protected function _toHtml()
     {
 
         $aboveThreshold = $this->quoteHelper->aboveThreshold($this->quoteHelper->getSessionQuote()->getGrandTotal());
+        $disabledSkus = !$this->quoteHelper->checkDisabledForSku('',true);
```
Comment (dfroneberger) [inline Block/Checkout/Overview.php]: this function can be changed to be like the M1 one

File: Block/Checkout/Overview.php [hunk 2]
```diff
@@ -117,8 +118,9 @@ protected function _toHtml()
...
 
-        if ($this->quoteHelper->isEnabledOnCOP() && $aboveThreshold) {
+        if ($this->quoteHelper->isEnabledOnCOP() && $aboveThreshold && $disabledSkus) {
```
Comment (dfroneberger) [inline Block/Checkout/Overview.php]: the `disabledSkus` name seems like an array, so I think it should be changed to be a boolean type name here and elsewhere

File: etc/adminhtml/system.xml [hunk 3]
```diff
@@ -281,6 +281,10 @@
                         ]]>
                         </comment>
                     </field>
+                    <field id="disabled_skus" type="textarea" sortOrder="10" translate="label" showInStore="1" showInWebsite="1" showInDefault="1">
+                        <label>Disable bread checkout for skus</label>
+                        <comment><![CDATA[<div><p>Enter sku list separated by comma with skus of products for which bread checkout should not be used or available</p></div>]]></comment>
```
Comment (dfroneberger) [inline etc/adminhtml/system.xml]: Need to add an example and then capitalize Bread in the label and comment

File: Block/Checkout/Overview.php (hunk 2 above)
Comment (aleksa-andersinnovation) [reply Block/Checkout/Overview.php]: what would be a boolean type name?

Comment (dfroneberger) [reply Block/Checkout/Overview.php]: like isDisabledBySku or something like that

File: Helper/Quote.php [hunk 4]
```diff
@@ -486,4 +494,64 @@ public function validateAllowedProductTypes()
...
+
+    /**
+     * Check if for given sku bread checkout is disabled
+     *
+     * @param string $sku
+     * @param string $store
+     * @return bool
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function checkDisabledForSku($sku = null, $store = \Magento\Store\Model\ScopeInterface::SCOPE_STORE)
```
Comment (dfroneberger) [inline Helper/Quote.php]: could the $store parameter just be made a variable, since it's never actually called?

File: etc/adminhtml/system.xml [hunk 5]
```diff
@@ -281,6 +281,10 @@
                         ]]>
                         </comment>
                     </field>
+                    <field id="disabled_skus" type="textarea" sortOrder="10" translate="label" showInStore="1" showInWebsite="1" showInDefault="1">
+                        <label>Disable Bread checkout for skus</label>
+                        <comment><![CDATA[<div><p>Enter sku list separated by comma with skus of products for which bread checkout should not be used or available (example: <strong>sku1,sku2,sku3</strong>)</p></div>]]></comment>
```
Comment (dfroneberger) [inline etc/adminhtml/system.xml]: There's one more Bread to be capitalized in the comment here 👍

File: Helper/Quote.php (hunk 4 above)
Comment (aleksa-andersinnovation) [reply Helper/Quote.php]: I would leave it as is, since that is the way all other methods that fetch config values do the same.

**Pull request merged**